providers/PrivyProvider.tsx

```diff
@@ -0,0 +1,29 @@
+import { PrivyProvider as Provider } from '@privy-io/react-auth'
+import { ReactNode } from 'react'
+import { base } from 'viem/chains'
+
+const PrivyProvider = ({ children }: { children: ReactNode }) => {
+  return (
+    <Provider
+    appId={process.env.NEXT_PUBLIC_PRIVY_APP_ID || (() => {
```

Comment:
too much logic inline.
- please declare a new variable `privyAppId` where you handle the `error` handling logic